drivers/adc/adc_sam0.c

```diff
@@ -367,22 +380,44 @@ static int start_read(const struct device *dev,
 			return -EINVAL;
 		}
 
+#if defined(ADC_CTRLB_RESSEL)
 		adc->CTRLB.bit.RESSEL = ADC_CTRLB_RESSEL_8BIT_Val;
+#elif defined(ADC_CTRLC_RESSEL)
+		adc->CTRLC.bit.RESSEL = ADC_CTRLC_RESSEL_8BIT_Val;
+#else
+		return -EINVAL;
+#endif
 		break;
 	case 10:
 		if (sequence->oversampling) {
 			LOG_ERR("Oversampling requires 12 bit resolution");
 			return -EINVAL;
 		}
 
+#if defined(ADC_CTRLB_RESSEL)
 		adc->CTRLB.bit.RESSEL = ADC_CTRLB_RESSEL_10BIT_Val;
+#elif defined(ADC_CTRLC_RESSEL)
+		adc->CTRLC.bit.RESSEL = ADC_CTRLC_RESSEL_10BIT_Val;
+#else
+		return -EINVAL;
+#endif
 		break;
 	case 12:
+#if defined(ADC_CTRLB_RESSEL)
 		if (sequence->oversampling) {
 			adc->CTRLB.bit.RESSEL = ADC_CTRLB_RESSEL_16BIT_Val;
 		} else {
 			adc->CTRLB.bit.RESSEL = ADC_CTRLB_RESSEL_12BIT_Val;
 		}
+#elif defined(ADC_CTRLC_RESSEL)
+		if (sequence->oversampling) {
+			adc->CTRLC.bit.RESSEL = ADC_CTRLC_RESSEL_16BIT_Val;
+		} else {
+			adc->CTRLC.bit.RESSEL = ADC_CTRLC_RESSEL_12BIT_Val;
+		}
+#else
+		return -EINVAL;
+#endif
 
 		break;
```

Comment:
Drop empty line

Reply:
Again, not really part of the patch, but can do if you want.

Reply:
I prefer keep consistency in the code. Please, drop line 421.